File: src/main/java/com/yupi/airouter/model/entity/ApiKey.java
package com.yupi.airouter.model.entity;

import com.mybatisflex.annotation.Column;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import com.mybatisflex.core.keygen.KeyGenerators;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * API Key 实体类
 *
 * @author <a href="https://codefather.cn">编程导航学习圈</a>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("api_key")
public class ApiKey implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * id
     */
    @Id(keyType = KeyType.Generator, value = KeyGenerators.snowFlakeId)
    private Long id;

    /**
     * 用户id
     */
    @Column("userId")
    private Long userId;

    /**
     * API Key值（sk-xxx格式）
     */
    @Column("keyValue")
    private String keyValue;

    /**
     * Key名称/备注
     */
    @Column("keyName")
    private String keyName;

    /**
     * 状态：active/inactive/revoked
     */
    @Column("status")
    private String status;

    /**
     * 已使用Token总数
     */
    @Column("totalTokens")
    private Long totalTokens;

    /**
     * 最后使用时间
     */
    @Column("lastUsedTime")
    private LocalDateTime lastUsedTime;

    /**
     * 创建时间
     */
    @Column("createTime")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Column("updateTime")
    private LocalDateTime updateTime;

    /**
     * 是否删除
     */
    @Column(value = "isDelete", isLogicDelete = true)
    private Integer isDelete;
}

File: go-backend/internal/model/entity/api_key.go
package entity

import (
	"time"

	"gorm.io/plugin/soft_delete"
)

type ApiKey struct {
	ID           int64                  `json:"id,string" gorm:"column:id;primaryKey;autoIncrement"`
	UserID       int64                  `json:"userId,string" gorm:"column:userId"`
	KeyValue     string                 `json:"keyValue" gorm:"column:keyValue"`
	KeyName      string                 `json:"keyName" gorm:"column:keyName"`
	Status       string                 `json:"status" gorm:"column:status"`
	TotalTokens  int64                  `json:"totalTokens,string" gorm:"column:totalTokens"`
	LastUsedTime *time.Time             `json:"lastUsedTime" gorm:"column:lastUsedTime"`
	CreateTime   time.Time              `json:"createTime" gorm:"column:createTime"`
	UpdateTime   time.Time              `json:"updateTime" gorm:"column:updateTime"`
	IsDelete     soft_delete.DeletedAt  `json:"isDelete" gorm:"column:isDelete;softDelete:flag"`
}

func (ApiKey) TableName() string {
	return "api_key"
}
